### PR TITLE
Update Version20150224171107.php

### PR DESCRIPTION
**Pull request by rr-appadaptive**
Make this migration work with the latest MariaDB (10.4.x^) and MySql Versions.

**What I did**

**How I did it**

**How to verify it**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

### CODE REVIEW

File: Neos.Neos/Migrations/Mysql/Version20150224171107.php [hunk 1]
```diff
@@ -22,7 +22,6 @@ public function up(Schema $schema)
         $this->addSql("UPDATE typo3_neos_eventlog_domain_model_event e, (SELECT DISTINCT uid, persistence_object_identifier FROM typo3_neos_eventlog_domain_model_event) p SET e.parentevent = p.uid WHERE parentevent IS NOT NULL AND p.persistence_object_identifier = e.parentevent");
         $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event DROP persistence_object_identifier, CHANGE parentevent parentevent INT UNSIGNED DEFAULT NULL, CHANGE uid uid INT UNSIGNED AUTO_INCREMENT NOT NULL");
         $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event ADD CONSTRAINT FK_30AB3A75B684C08 FOREIGN KEY (parentevent) REFERENCES typo3_neos_eventlog_domain_model_event (uid)");
-        $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event ADD PRIMARY KEY (uid)");
```
Comment (albe) [inline Neos.Neos/Migrations/Mysql/Version20150224171107.php]: Could it be, that just the order is causing problems with mariadb 10.4? It tries to create the foreign key constraint before the primary key.

**Comment by Sebobo** [comment]
It seems it's not allowed anymore to add an index with the name of an existing `UNIQUE` index. At the point of the migration the `UNIQUE` index for `uid` already exists because it's set to auto increment.

What works is to temporarily rename it, add the intended index and then remove the old renamed index like this:

```
$this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event DROP INDEX uid, ADD INDEX uidtemp (uid)");
$this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event ADD PRIMARY KEY (uid)");
$this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event DROP INDEX uidtemp");
```

Sadly I couldn't find the change that caused this problem in the MariaDB updates.

This also makes #2666 unnecessary.

**Comment by gjwnc** [comment]
I just stumbled upon this problem too and it seems to be related to this mariadb bug https://jira.mariadb.org/browse/MDEV-19598. I too, didn't find much further details regarding this MariaDB bug.

**Comment by albe** [comment]
> $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event DROP INDEX uid, ADD INDEX uidtemp (uid)");
> $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event ADD PRIMARY KEY (uid)");
> $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event DROP INDEX uidtemp");

Why would you first recreate the unique index with a different name, just to drop it afterwards anyway? The thing is, a primary key index is always unique. So it should be okay to just drop the unique index on `(uid)` before adding the primary key.

> $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event DROP INDEX uid");
> $this->addSql("ALTER TABLE typo3_neos_eventlog_domain_model_event ADD PRIMARY KEY (uid)");

**Comment by Sebobo** [comment]
@albe that doesn't work with the specified MariaDB versions. I tried that. You get an error that you cannot drop the index.

**Comment by albe** [comment]
Ugh, okay. Then let's go that route to fix things for 10.4

**Comment by Sebobo** [comment]
I'll create a separate PR with my proposal that targets Neos 3.3

**Pull request closed without merging**